common/mapper/src/main/java/io/helidon/common/mapper/Value.java

```diff
@@ -0,0 +1,296 @@
+     * return an {@code Optional} describing the value, otherwise return an
+     * empty {@code Optional}.
+     * <p>
+     * Copied from {@link Optional}. You can get real optional from {@link #asOptional()}.
+     *
+     * @param predicate a predicate to apply to the value, if present
+     * @return an {@code Optional} describing the value of this {@code Optional}
+     * if a value is present and the value matches the given predicate,
+     * otherwise an empty {@code Optional}
+     * @throws NullPointerException if the predicate is null
+     * @see Optional#filter(java.util.function.Predicate)
+     */
+    default Optional<T> filter(Predicate<? super T> predicate) {
+        return asOptional().filter(predicate);
+    }
+
+    /**
+     * If a value is present, apply the provided mapping function to it,
+     * and if the result is non-null, return an {@code Optional} describing the
+     * result.  Otherwise return an empty {@code Optional}.
+     *
+     * @param <U>    The type of the result of the mapping function
+     * @param mapper a mapping function to apply to the value, if present
+     * @return an {@code Optional} describing the result of applying a mapping
+     * function to the value of this {@code Optional}, if a value is present,
+     * otherwise an empty {@code Optional}
+     * @throws NullPointerException if the mapping function is null
+     *
+     *                              <p>
+     *                              Copied from {@link Optional}. You can get real optional from {@link #asOptional()}.
+     * @see Optional#map(Function)
+     */
+    default <U> Optional<U> map(Function<? super T, ? extends U> mapper) {
+        return asOptional().map(mapper);
+    }
+
+    /**
+     * If a value is present, apply the provided {@code Optional}-bearing
+     * mapping function to it, return that result, otherwise return an empty
+     * {@code Optional}.  This method is similar to {@link #map(Function)},
+     * but the provided mapper is one whose result is already an {@code Optional},
+     * and if invoked, {@code flatMap} does not wrap it with an additional
+     * {@code Optional}.
+     *
+     * <p>
+     * Copied from {@link Optional}. You can get real optional from {@link #asOptional()}.
+     *
+     * @param <U>    The type parameter to the {@code Optional} returned by
+     * @param mapper a mapping function to apply to the value, if present
+     *               the mapping function
+     * @return the result of applying an {@code Optional}-bearing mapping
+     * function to the value of this {@code Optional}, if a value is present,
+     * otherwise an empty {@code Optional}
+     * @throws NullPointerException if the mapping function is null or returns
+     *                              a null result
+     * @see Optional#flatMap(Function)
+     */
+    default <U> Optional<U> flatMap(Function<? super T, Optional<U>> mapper) {
```

Comment:
Why not `Function<? super T, Optional<? extends U>> mapper` ?

Reply:
fixed